File: Glimpse.Player/Codecs/Flac/FlacCodec.cs
using System.IO;

namespace Glimpse.Player.Codecs.Flac;

public class FlacCodec : Codec
{
    public override bool FileIsSupported(string path, string extension)
    {
        return extension == ".flac";
    }

    public override CodecStream CreateStream(string path)
    {
        return new FlacStream(path);
    }
}

File: src/Player/Codecs/FLACCodec.cpp
#include "FLACCodec.h"

#include "../Common.h"
#include <Slant++/Stream/Flac.h>

#include <algorithm>

namespace gmp
{
    std::unique_ptr<sls::AudioStream> FLACCodec::CreateStream(const std::filesystem::path& path)
    {
        return std::make_unique<sls::Flac>(path);
    }

    bool FLACCodec::CheckFileSupport(const std::filesystem::path& path)
    {
        auto ext = path.extension().string();
        std::ranges::transform(ext, ext.begin(), tolower);
        return path.extension() == ".flac";
    }
}
